src/pudl/transform/ferc1.py

```diff
@@ -2477,6 +2573,54 @@ def spot_fix_rows(self, df: pd.DataFrame) -> pd.DataFrame:
         return df
 
 
+class ElectricEnergyAccountSourcesTransformer(Ferc1AbstractTableTransformer):
+    """Transformer class for :ref:`electric_energy_account_sources_ferc1` table.
+
+    For XBRL, this is a duration-only table. Right now we are merging in the metadata
+    but not actually keeping anything from it. We are also not yet doing anything with
+    the sign.
+    """
+
+    table_id: Ferc1TableId = Ferc1TableId.ELECTRIC_ENERGY_ACCOUNT_SOURCES_FERC1
+    has_unique_record_ids: bool = False
+
+    def normalize_metadata_xbrl(
+        self, xbrl_fact_names: list[str] | None
+    ) -> pd.DataFrame:
+        """Normie.
+
+        Determine whether the
+        """
+        eeas_meta = (
+            super().normalize_metadata_xbrl(xbrl_fact_names)
+            # should this be a straight up rename? could we just rename it within the
+            # standard normalize_metadata_xbrl??
+            .assign(xbrl_factoid=lambda x: x.xbrl_fact_name)
+        )
+        # # Flag the metadata record types
+        eeas_meta.loc[
+            eeas_meta.calculations.astype(bool), "row_type_xbrl"
+        ] = "calculated"
+        eeas_meta.loc[
+            ~eeas_meta.calculations.astype(bool),
+            "row_type_xbrl",
+        ] = "energy_source"
+        # Save the normalized metadata so it can be used by other methods.
+        self.xbrl_metadata_normalized = eeas_meta
+        return eeas_meta
+
+    @cache_df(key="dbf")
+    def process_dbf(self, raw_dbf: pd.DataFrame) -> pd.DataFrame:
+        """Start with inherited method and do some energy source-specific processing.
```

Comment:
Note that we're splitting this input table into two tables with two different transforms -- and we're dropping all the columns that will go in the other table.